main.go

```diff
@@ -570,12 +594,20 @@ func ProcessDirEntry(s *State, word string, resultChan chan<- Result) {
 }
 
 func PrintDnsResult(s *State, r *Result) {
+	output := ""
 	if r.Status == 404 {
-		fmt.Printf("Missing: %s\n", r.Entity)
+		output += fmt.Sprintf("Missing: %s\n", r.Entity)
 	} else if s.ShowIPs {
-		fmt.Printf("Found: %s [%s]\n", r.Entity, r.Extra)
+		output += fmt.Sprintf("Found: %s [%s]\n", r.Entity, r.Extra)
+	} else if s.ShowCNAME {
+		output += fmt.Sprintf("Found: %s [%s]\n", r.Entity, r.Extra)
 	} else {
-		fmt.Printf("Found: %s\n", r.Entity)
+		output += fmt.Sprintf("Found: %s\n", r.Entity)
+	}
+	fmt.Printf(output)
```

Comment:
Actually, I did some reading and it looks like `fmt.Println` will not work here as it doesn't return `string`, so we won't be able to save the value to a variable here. `fmt.Println` actually returns 2 values - `int` of number of bytes written and `err` - [https://golang.org/pkg/fmt/#Println](https://golang.org/pkg/fmt/#Println).

My quick Google-fu did not yield any simple solutions without using `fmt.Sprintf`, may need to go with this ugliness below instead?
`output += "Found: " + r.Entity + "\n"`